File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        # find the largest volumn in a greedy way
        l = 0
        r = len(height) - 1
        max_vol = 0
        while l < r:
            max_vol = max(min(height[l], height[r]) * (r - l), max_vol)
            if height[l] < height[r]:
                l+=1
            else:
                r-=1
        return max_vol

File: 0011-container-with-most-water/0011-container-with-most-water.ts
function maxArea(height: number[]): number {
    
    let res= 0, l = 0, r =height.length-1;

    while (l < r){
        res = Math.max(res,Math.min(height[l], height[r]) * (r-l))
        height[l] >= height[r] ? r--: l++
    }
    return res

};